kernel/init.c

```diff
@@ -355,6 +355,8 @@ static void bg_thread_main(void *unused1, void *unused2, void *unused3)
 #ifdef CONFIG_COVERAGE_DUMP
 	/* Dump coverage data once the main() has exited. */
 	gcov_coverage_dump();
+#elif defined(CONFIG_COVERAGE_SEMIHOST)
+	gcov_coverage_semihost();
```

Comment:
This should also be added at the other default location for `gcov_coverage_dump` in `reboot.c`.